### PR TITLE
pull request

### PR DESCRIPTION
**Pull request by Hnguyen1997**
migration warning of migration is applied before its dependency on database 'default' when run

Can add technologies when edit project
Can view other course members' schedule including instructor
Can schedule meeting when click on other people schedule. Cannot change that person busy time or change other people scheduled time. Can hover over the event to see if the scheduled meeting time belong to you.
Send alert to the person schedule

### CODE REVIEW

File: teamwork/apps/profiles/models.py [hunk 1]
```diff
@@ -50,6 +50,38 @@ def save(self, *args, **kwargs):
         """
         super(Skills, self).save(*args, **kwargs)
 
+#-----------------------------------------------------------------------
+class Techs(models.Model):
```
Comment (Kapn) [inline teamwork/apps/profiles/models.py]: Can you move this Techs model from `apps/profiles/models.py` to `apps/projects/models.py`

File: teamwork/apps/profiles/models.py [hunk 2]
```diff
@@ -204,6 +236,11 @@ class Profile(models.Model):
     known_skills = models.ManyToManyField(Skills, related_name="known", default="")
     learn_skills = models.ManyToManyField(Skills, related_name="learn", default="")
 
+#------------------------------------------------------------------------------------
+    known_techs = models.ManyToManyField(Techs, related_name="known", default="")
```
Comment (Kapn) [inline teamwork/apps/profiles/models.py]: Need to change the related name since `known` and `learn` are already used by skills. Please change related_name to `known_technologies` and `learn_technologies`

File: teamwork/apps/profiles/views/EditProfileView.py [hunk 3]
```diff
@@ -157,19 +175,18 @@ def edit_profile_helper(request, username):
         Helper function that saves profile information from the ProfileForm
     """
```
Comment (Kapn) [inline teamwork/apps/profiles/views/EditProfileView.py]: Please revert the below change. Add a `tempProfile` to the else: instead of pulling outside.

**Review by Kapn — CHANGES_REQUESTED** [review]
See comments

File: teamwork/apps/profiles/views/EditProfileView.py [hunk 4]
```diff
@@ -157,19 +175,18 @@ def edit_profile_helper(request, username):
...
 
-    if request.user.profile.isGT:
-        tempProfile = User.objects.get(username=username)
-        profile = Profile.objects.get(user=tempProfile)
-    else:
+    
         #grab profile for the current user
-        profile = Profile.objects.get(user=request.user)
+    tempProfile = User.objects.get(username=username)
+    profile = Profile.objects.get(user=tempProfile)
```
Comment (Kapn) [inline teamwork/apps/profiles/views/EditProfileView.py]: Please revert this change. Instead, add a `tempProfile` to the else.

File: teamwork/apps/profiles/views/EditProfileView.py [hunk 5]
```diff
@@ -163,13 +181,17 @@ def edit_profile_helper(request, username):
     else:
         #grab profile for the current user
         profile = Profile.objects.get(user=request.user)
+        tempProfile = User.objects.get(username=username)
```
Comment (rmonroe) [inline teamwork/apps/profiles/views/EditProfileView.py]: Change this line to 

```profileUser = User.objects.get(username=profile.username)```

Username is the page user and as a result, you would be able to change any users email that is why it is in the Admin(isGT) if

File: teamwork/apps/profiles/views/EditProfileView.py [hunk 6]
```diff
@@ -178,6 +200,8 @@ def edit_profile_helper(request, username):
         #if data is entered, save it to the profile for the following
         if name:
             profile.name = name
+        if email:
+            tempProfile.email=email
```
Comment (rmonroe) [inline teamwork/apps/profiles/views/EditProfileView.py]: Based on above

```profileUser.email = email```

File: teamwork/apps/projects/views/EditProjectView.py [hunk 7]
```diff
@@ -115,7 +115,7 @@ def edit_project(request, slug):
         elif profAdded:
             messages.add_message(request, messages.SUCCESS, "Greppers have been added to the project.")
         else:
-            messages.add_message(request, messages.WARNING, "Failed to invite member(s) to project. Make sure they are enrolled in this course.")
+            messages.add_message(request, messages.WARNING, "Student(s) is already added to the project.")
```
Comment (rmonroe) [inline teamwork/apps/projects/views/EditProjectView.py]: This message will need a check to see what failed since this message can also still be triggered if the user is not enrolled in the course

File: teamwork/templates/projects/create_project.html [hunk 8]
```diff
@@ -65,6 +59,18 @@
           page: params.page
         };
       },
+       
+  // $(".s2_desired_techs").select2({
```
Comment (rmonroe) [inline teamwork/templates/projects/create_project.html]: Is the select 2 function not setup for technologies?

File: teamwork/templates/projects/create_project.html [hunk 9]
```diff
@@ -152,7 +156,7 @@ <h3 class="box-title">Create Project</h3>
         <div class="form-group">
             {{ form.content.errors }}
             <label for="{{ form.content.id_for_label }}" class="control-label">
-            Project description <a href="https://guides.github.com/features/mastering-markdown/">[Markdown supported]</a>:
+            Project description <a href="https://daringfireball.net/projects/markdown/syntax">[Markdown supported]</a>:
```
Comment (rmonroe) [inline teamwork/templates/projects/create_project.html]: Revert this back to the github  link

File: teamwork/templates/projects/create_project.html [hunk 10]
```diff
@@ -215,16 +228,6 @@ <h3 class="box-title">Optional Settings</h3>
             {{ form.slug }}
         </div>
 
-        <div class="form-group">
```
Comment (rmonroe) [inline teamwork/templates/projects/create_project.html]: Revert this back as well

**Review by rmonroe — CHANGES_REQUESTED** [review]

File: teamwork/templates/projects/create_project.html (hunk 8 above)
Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: its in templates/projects/edit_project.html

File: teamwork/templates/projects/create_project.html (hunk 9 above)
Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: Okay

File: teamwork/templates/projects/create_project.html (hunk 10 above)
Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: We already have this in that file, can you elaborate if something else needs to be done.

File: teamwork/apps/projects/views/EditProjectView.py (hunk 7 above)
Comment (anishatavva) [reply teamwork/apps/projects/views/EditProjectView.py]: I think we added an elif statement before but it was giving us and error, is there another approach we should use

Comment (rmonroe) [reply teamwork/apps/projects/views/EditProjectView.py]: I would recommend just putting another if within the else that determines which message to send

File: teamwork/templates/projects/create_project.html (hunk 8 above)
Comment (rmonroe) [reply teamwork/templates/projects/create_project.html]: Yes, but is it not a part of project creation like skills?

File: teamwork/templates/projects/create_project.html (hunk 10 above)
Comment (rmonroe) [reply teamwork/templates/projects/create_project.html]: This section appears to be removed by your PR and we do not want that. This was a feature from our update and probably just got mixed up in a merge

Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: We already have this in our version of the code, maybe we will just re push the branch when we are done with making changes to see if it shown in the code

File: teamwork/templates/projects/create_project.html (hunk 8 above)
Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: We've put in s2 function for technologies in the same create_project.html before but it gave us an error. We have the select 2 function for technologies in edit_project.html

File: teamwork/apps/projects/views/EditProjectView.py (hunk 7 above)
Comment (rmonroe) [reply teamwork/apps/projects/views/EditProjectView.py]: I would put a new if inside of the else that determines which message to send

File: teamwork/templates/projects/create_project.html (hunk 8 above)
Comment (rmonroe) [reply teamwork/templates/projects/create_project.html]: Okay, what was the error? Just cause something is causing an error doesn't mean it should just be avoided

File: teamwork/templates/projects/create_project.html (hunk 10 above)
Comment (rmonroe) [reply teamwork/templates/projects/create_project.html]: Yeah if this will still appears in the code at the end then that is fine, I just want to make sure this feature doesn't get lost

File: teamwork/templates/projects/create_project.html (hunk 8 above)
Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: When we put in the s2 function for technologies in the create_project, the technologies that we add are not being displayed in the ui

Comment (anishatavva) [reply teamwork/templates/projects/create_project.html]: Ooops our bad, it worked, we must have missed it last time!

File: teamwork/apps/profiles/views/EditProfileView.py [hunk 11]
```diff
@@ -30,6 +31,23 @@ def edit_skills(request, username):
 
     return HttpResponse("Failure")
 
+def edit_techs(request, username):
```
Comment (rmonroe) [inline teamwork/apps/profiles/views/EditProfileView.py]: This needs to be removed from profiles

File: teamwork/apps/courses/views/EditCourseView.py [hunk 12]
```diff
@@ -74,7 +74,7 @@ def edit_course(request, slug):
         if added:
             messages.add_message(request, messages.SUCCESS, "Successfully added member(s) to course.")
         else:
-            messages.add_message(request, messages.SUCCESS, "Failed to successfully add member(s) to course.")
```
Comment (rmonroe) [inline teamwork/apps/courses/views/EditCourseView.py:77]: Why was this message changed?

File: teamwork/templates/adminlte/lib/_main_header.html [hunk 13]
```diff
@@ -98,24 +112,31 @@
...
-                                    </button>
-                                </a>
-                                <a href="{% block logout_url %}{% logout_url %}{% endblock %}">
-                                    <button type="button" class="btn btn-default" style="background: rgba(255, 0, 0, 0.7);">
-                                        <font color="white">Sign Out</font>
-                                    </button>
-                                </a>
+                              <a href="{% url 'profile' user.username %}">
+							  <!-- need to add margin:2px 0 for Light/Dark theme -->
+                                  <button type="button" class="btn btn-flat" style="background: rgba(243, 156, 18, 1);margin:2px 0">
```
Comment (rmonroe) [inline teamwork/templates/adminlte/lib/_main_header.html]: use class btn btn-default

File: teamwork/templates/projects/view_projects.html [hunk 14]
```diff
@@ -3,17 +3,8 @@
 {% load static %}
 {% load i18n %}
 
-{% block stylesheets %}
-
```
Comment (rmonroe) [inline teamwork/templates/projects/view_projects.html]: Why have the tooltips been removed?

**Review by rmonroe — CHANGES_REQUESTED** [review]

File: teamwork/templates/projects/partial_project_profile.html [hunk 15]
```diff
@@ -54,7 +62,7 @@ <h5 class="widget-user-desc">{{project.tagline}}</h5>
     {% if forloop.last %}</div><div class="clearfix"></div>{% endif %}
 {% empty %}
   <div>
-  </h3>You are not in any projects</h3>
```
Comment (rmonroe) [inline teamwork/templates/projects/partial_project_profile.html]: This should be you since you are addressing the user

**Review by rmonroe — CHANGES_REQUESTED** [review]

**Comment by rmonroe** [comment]
Also the following commits need to be removed from the log

https://github.com/Hnguyen1997/grepthink/pull/3/commits/7f664883454213c24255e67123fc94b5
https://github.com/Hnguyen1997/grepthink/pull/3/commits/3ecf0039378b2933c56ef989a4b6a64

The first commits removes all the files from the repo and the second puts them all back.

I believe KP plans to handle this personally when we go ahead with the merge.

File: teamwork/apps/profiles/views/EditProfileView.py (hunk 5 above)
Comment (Kapn) [reply teamwork/apps/profiles/views/EditProfileView.py]: `tempProfile = User.objects.get(username=username)` will allow any user to visit another user's profile and edit their email address. This should be `tempProfile = User.objects.get(username=profile.username)`.

**Review by Kapn — DISMISSED** [review]

File: teamwork/templates/projects/view_projects.html (hunk 14 above)
Comment (shreddy1) [reply teamwork/templates/projects/view_projects.html]: We never touched this so that is weird, but should we put it back in?

File: teamwork/templates/projects/partial_project_profile.html (hunk 15 above)
Comment (shreddy1) [reply teamwork/templates/projects/partial_project_profile.html]: Wait what do you mean, can you clarify please?

File: teamwork/apps/courses/views/EditCourseView.py (hunk 12 above)
Comment (shreddy1) [reply teamwork/apps/courses/views/EditCourseView.py:77]: It wasn't we were trying to fix that task so we were just playing around with it. But we fixed this task that you assigned for us on the PR.

File: teamwork/templates/projects/view_projects.html (hunk 14 above)
Comment (shreddy1) [reply teamwork/templates/projects/view_projects.html]: https://github.com/Hnguyen1997/grepthink/blob/master/teamwork/templates/projects/view_projects.html 

If you look at this which I think is your version of it, it never had this in the code! Let us know what you want us to do?

File: teamwork/apps/courses/views/EditCourseView.py (hunk 12 above)
Comment (rmonroe) [reply teamwork/apps/courses/views/EditCourseView.py:77]: Apologies, this message is correct

File: teamwork/templates/projects/partial_project_profile.html (hunk 15 above)
Comment (rmonroe) [reply teamwork/templates/projects/partial_project_profile.html]: The message appears to have been changed to They but in our version it is You and You is the correct version since it it meant to be addressing the user

File: teamwork/templates/projects/view_projects.html (hunk 14 above)
Comment (rmonroe) [reply teamwork/templates/projects/view_projects.html]: No that Url is from your fork, our version has the lines removed added since I personally worked on this task on our branch. Please add the removed lines back to your branch as well, I believe they may have been removed during a merge conflict updating your fork with ours

**Comment by rmonroe** [comment]
Please remove: 
https://github.com/grepthink/grepthink/blob/aaf72137a76522a026e0b1ac27472a9d469b388b/docs/Sprint2zcwfeature/LightDarktheme.docx

https://github.com/grepthink/grepthink/blob/aaf72137a76522a026e0b1ac27472a9d469b388b/docs/Sprint2zcwfeature/migration%20bug%20fixed.docx

Since we do not need word doc explaining how to insert code in the repo

File: teamwork/apps/profiles/forms.py [hunk 16]
```diff
@@ -111,6 +111,10 @@ def __init__(self, *args, **kwargs):
     name = forms.CharField(
               widget=forms.TextInput(attrs={'class': 'form-control'}),
               max_length=75, required=False)
+
```
Comment (rmonroe) [inline teamwork/apps/profiles/forms.py:114]: Add the email validator from profiles/forms.py/SignupForm

File: teamwork/apps/projects/views/EditProjectView.py [hunk 17]
```diff
@@ -115,7 +115,10 @@ def edit_project(request, slug):
         elif profAdded:
             messages.add_message(request, messages.SUCCESS, "Greppers have been added to the project.")
         else:
-            messages.add_message(request, messages.WARNING, "Failed to invite member(s) to project. Make sure they are enrolled in this course.")
+            if (not course in mem_courses):
+                messages.warning(request, "User failed to be added to the project. " + mem_to_add.username + " is not enrolled in the course")
+                # return False
+            #messages.add_message(request, messages.WARNING, "Student(s) is already added to the project.")
```
Comment (rmonroe) [inline teamwork/apps/projects/views/EditProjectView.py]: Put the message stating the student is already in the project in the else on this inner if

File: teamwork/templates/adminlte/lib/_main_header.html [hunk 18]
```diff
@@ -98,24 +112,31 @@
...
-                                </a>
+                              <a href="{% url 'profile' user.username %}">
+							  <!-- need to add margin:2px 0 for Light/Dark theme -->
+                                  <button type="button" class="btn btn-default" style="background: rgba(243, 156, 18, 1);margin:2px 0">
+                                      <font color="white">Profile</font>
+                                  </button>
+                              </a>
+							  <!-- need to add margin:2px 0 for Light/Dark theme -->
+                              <a href="{% url 'edit_schedule' user.username %}">
+                                  <button type="button" class="btn btn-flat" style="background: rgba(0, 150, 255, 1);margin:2px 0">
```
Comment (rmonroe) [inline teamwork/templates/adminlte/lib/_main_header.html]: use btn btn-default

File: teamwork/templates/adminlte/lib/_main_header.html [hunk 19]
```diff
@@ -98,24 +112,31 @@
...
+                                      <font color="white">Edit Schedule</font>
+                                  </button>
+                              </a>
+							  <!-- main code of the Light/Dark theme switch -->
+							  <a href="javascript:switchSkin()">
+                                  <button type="button" class="btn btn-flat  btn-info" style="margin:2px 0">
```
Comment (rmonroe) [inline teamwork/templates/adminlte/lib/_main_header.html]: use btn btn default

File: teamwork/templates/adminlte/lib/_main_header.html [hunk 20]
```diff
@@ -98,24 +112,31 @@
...
+                                      <font color="white">Switch Theme</font>
+                                  </button>
+                              </a>
+							  <!-- add style="display:block;clear:both;" and margin:2px 0;width:100%; for Light/Dark theme -->
+                              <a href="{% block logout_url %}{% logout_url %}{% endblock %}"style="display:block;clear:both;">
+                                  <button type="button" class="btn btn-flat" style="background: rgba(255, 0, 0, 0.7);margin:2px 0;width:100%;">
```
Comment (rmonroe) [inline teamwork/templates/adminlte/lib/_main_header.html]: use btn btn default

File: teamwork/templates/projects/create_project.html (hunk 10 above)
Comment (rmonroe) [reply teamwork/templates/projects/create_project.html]: PUT THIS BACK

Comment (rmonroe) [reply teamwork/templates/projects/create_project.html]: PUT THIS BACK You removed it in this commit https://github.com/grepthink/grepthink/pull/238/commits/5f32501acd0db774c359fbc3dbad23afaf0b6dfd

**Review by rmonroe — APPROVED** [review]
